src/libraries/Microsoft.PowerFx.Core/Public/Config/SymbolTable.cs

```diff
@@ -0,0 +1,388 @@
+            else if (_environmentSymbolDisplayNameProvider.TryGetLogicalName(name, out var logicalName))
+            {
+                lookupName = logicalName;
+                displayName = name;
+            }
+
+            return _environmentSymbols.TryGetValue(lookupName, out symbol);
+        }
+
+        // $$$ derived symbol tables can hook. 
+        internal virtual bool TryLookup(DName name, out NameLookupInfo nameInfo)
+        {
+            nameInfo = default;
+            return false;
+        }
+
+        bool INameResolver.Lookup(DName name, out NameLookupInfo nameInfo, NameLookupPreferences preferences)
+        {
+            if (TryLookup(name, out nameInfo))
+            {
+                return true;
+            }
+
+            if (_variables.TryGetValue(name.Value, out nameInfo))
+            {
+                return true;
+            }
+
+            if (TryGetSymbol(name, out var symbol, out var displayName))
+            {
+                // Special case symbols
+                if (symbol is IExternalOptionSet optionSet)
+                {
+                    nameInfo = new NameLookupInfo(
+                        BindKind.OptionSet,
+                        optionSet.Type,
+                        DPath.Root,
+                        0,
+                        optionSet,
+                        displayName);
+
+                    return true;
+                }
+                else
+                {
+                    throw new NotImplementedException($"{symbol.GetType().Name} not supported.");
+                }
+            }
+
+            var enumValue = GetEnumSymbolSnapshot.FirstOrDefault(symbol => symbol.InvariantName == name);
+            if (enumValue != null)
+            {
+                nameInfo = new NameLookupInfo(BindKind.Enum, enumValue.EnumType, DPath.Root, 0, enumValue);
+                return true;
+            }
+
+            nameInfo = default;
+            return false;
+        }
+
+        bool INameResolver.TryGetInnermostThisItemScope(out NameLookupInfo nameInfo)
+        {
+            nameInfo = default;
+            return false;
+        }
+
+        bool INameResolver.LookupDataControl(DName name, out NameLookupInfo lookupInfo, out DName dataControlName)
+        {
+            dataControlName = default;
+            lookupInfo = default;
+            return false;
+        }
+
+        IEnumerable<TexlFunction> INameResolver.LookupFunctions(DPath theNamespace, string name, bool localeInvariant)
+        {
+            Contracts.Check(theNamespace.IsValid, "The namespace is invalid.");
+            Contracts.CheckNonEmpty(name, "name");
+
+            // See TexlFunctionsLibrary.Lookup
+            // return _functionLibrary.Lookup(theNamespace, name, localeInvariant, null);            
+            var functionLibrary = _functions.Where(func => func.Namespace == theNamespace && name == (localeInvariant ? func.LocaleInvariantName : func.Name)); // Base filter
```

Comment:
>func.Namespace == theNamespace && name == (localeInvariant ? func.LocaleInvariantName : func.Name) [](http://example.com/codeflow?start=59&length=98)

equality should be done in functions that know if we are case sensitive or not

Reply:
this is just a copy of existing code.